src/uniq.js

```diff
@@ -18,4 +20,39 @@ var uniqWith = require('./uniqWith');
  *      R.uniq([1, '1']);     //=> [1, '1']
  *      R.uniq([[42], [42]]); //=> [[42]]
  */
-module.exports = uniqWith(equals);
+module.exports = _curry1(function uniq(list) {
+  /* global Set */
+  var item, set, idx, len, items, uniqs;
+
+  /**
+   * Uses a native `Set` instance where possible for
+   * removing duplicate items. Items that implement
+   * the fantasy-land Setoid spec fallback to using
+   * `_contains` to support custom equality behaviour.
+   */
+  if (typeof Set === 'function') {
```

Comment:
I was writing a similar build script to modularize our files at the same time David did his, and my output was much uglier because I ended up with those IIFEs.  I took forever to build mine because I spent all sorts of effort trying to determine the cases where I could skip these IIFEs.  David's approach has been much simpler.  And there have only been a handful of cases where it's caused us any difficulty at all.